raglweb/forfeit_games.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from sqlite3 import Connection
+
+
+def get_player_forfeit_wins(db: Connection, player_id: int) -> int:
+    res = db.execute(f"SELECT COUNT(*) "
+                     f"FROM forfeit_games "
+                     f"WHERE profile_id0={player_id}").fetchone()[0]
+    return res
+
+
+def get_player_forfeit_losses(db: Connection, player_id: int) -> int:
+    res = db.execute(f"SELECT COUNT(*) "
+                     f"FROM forfeit_games "
+                     f"WHERE profile_id1={player_id}").fetchone()[0]
+    return res
+
+
+def get_player_forfeit_games(db: Connection, player_id: int) -> dict:
+    query = f"SELECT fg.profile_id0, (SELECT profile_name FROM players WHERE profile_id=fg.profile_id0) as p0_name, " \
+            f"fg.profile_id1, (SELECT profile_name FROM players WHERE profile_id=fg.profile_id1) as p1_name, " \
```

Comment:
Not a blocking comment nested select sounds pretty expensive. Similarly, running a new query in a loop (the other 2 functions are called like that) is going to be quite a hit on the server

Reply:
I replaced the 2 other functions you mention with one that just grabs forfeit games win/loss stats for all players. Left the nested query as is though for lack of good ideas on how to change it.